apps/condo/domains/resident/tasks/sendBillingReceiptsAddedNotifications.js

```diff
@@ -0,0 +1,165 @@
+const pino = require('pino')
+const dayjs = require('dayjs')
+const falsey = require('falsey')
+const { get, isEmpty, isUndefined } = require('lodash')
+
+const conf = require('@core/config')
+const { getSchemaCtx } = require('@core/keystone/schema')
+
+const { getStartDates, getLoadByChunks } = require('@condo/domains/common/utils/task/helpers')
+const { RedisVar } = require('@condo/domains/common/utils/redis-var')
+
+const { BillingIntegrationOrganizationContext, BillingReceipt, BillingCategory } = require('@condo/domains/billing/utils/serverSchema')
+
+const { sendMessage } = require('@condo/domains/notification/utils/serverSchema')
+
+const {
+    BILLING_RECEIPT_ADDED_WITH_DEBT_TYPE,
+    BILLING_RECEIPT_ADDED_WITH_NO_DEBT_TYPE,
+} = require('@condo/domains/notification/constants/constants')
+
+const { Resident, ServiceConsumer } = require('@condo/domains/resident/utils/serverSchema')
+const { COUNTRIES, DEFAULT_LOCALE } = require('@condo/domains/common/constants/countries')
+const { CURRENCY_SYMBOLS, DEFAULT_CURRENCY_CODE } = require('@condo/domains/common/constants/currencies')
+
+const {
+    REDIS_LAST_SEND_BILLING_RECEIPT_NOTIFICATION_CREATED_AT_NAME,
+    REDIS_SEND_BILLING_RECEIPT_NOTIFICATION_LOCKED_AT_NAME,
+} = require('../constants')
+
+const logger = pino({
+    name: 'send_billing_receipt_available_notifications',
+    enabled: falsey(process.env.DISABLE_LOGGING),
+})
+
+const MAX_NOTIFICATIONS_ALLOWED = 1000
+const LOCK_EXPIRE_SEC = 10 * 60
+
+const getLoadOneCached = (context) => {
+    const cache = {}
+
+    return async (list, where) => {
+        const listName = list.gql.SINGULAR_FORM
+        const key = JSON.stringify(where)
+        if (isUndefined(cache[listName])) cache[listName] = {}
+        if (!isUndefined(cache[listName][key])) return cache[listName][key]
+
+        cache[listName][key] = await list.getOne(context, { ...where, deletedAt: null })
+
+        return cache[listName][key]
+    }
+}
+
+const sendBillingReceiptsAddedNotifications = async (forceSend = false, _maxNotificationsCount = MAX_NOTIFICATIONS_ALLOWED, forceUnlock = false, resendFromDt = null) => {
```

Comment:
It's too strange why do you really need so many arguments? why do you need `maxCount`? it's really strange

Reply:
This script can be executed automatically by cron or manually. This limit allows to proceed receipts by portions in order to use resources in more optimal way.